File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> vals = new ArrayList<Integer>();
        
        TreeNode curr = root;
        
        DFS(curr, vals);
        
        return vals;
    }
    
    public void DFS(TreeNode node, List<Integer> list) {
        
        if (node == null) {
            return;
        }
        
   
        DFS(node.left, list);
        list.add(node.val);
        DFS(node.right, list);
        
        
    }
}

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root):
        lst = []
        stack = []
        cur = root
        
        while(len(stack) or cur):
            while(cur):
                stack.append(cur)
                cur = cur.left
            
            popped = stack.pop()
            lst.append(popped.val)
            cur = popped.right
        
        return lst